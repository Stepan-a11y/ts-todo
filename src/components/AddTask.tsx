import React, { useState } from 'react';
import { useAction } from '../hooks/useAction';
import './styles/AddTask.css';

const AddTask: React.FC = () => {
  const { addNewTodo } = useAction();

  const [title, setTitle] = useState('');
  const [errorCount, setErrorCount] = useState(1);

  const addNewTask = () => {
    addNewTodo(title, false, false, false, new Date().toLocaleString());
    setTitle('');
  };

  const text = (e: React.ChangeEvent<HTMLInputElement>) => {
    setTitle(e.target.value);
    setErrorCount(-160 + e.target.value.length);
  };

  return (
    <>
      <div className='cont'>
        <input className='inp' type="text" value={title} onChange={(e) => text(e)} />
        <input className='but' disabled={title.length < 1} type="button" value="Добавить" onClick={addNewTask} />
      </div>
      {(title.length > 160) && <p className='err' style={{ color: "red" }}>Превышен лимит символов на {errorCount}</p>}
    </>
  );
}

export default AddTask;
