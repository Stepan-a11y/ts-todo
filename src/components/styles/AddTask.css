.cont {
    display: flex;
    flex-direction: row;
    width: 100%;
    height: 45px;
    margin-top: 80px;
}

.inp {
    width: 80%;
    padding-left: 20px;
    font-size: 18px;
    border: 3px solid rgb(209, 143, 245);
    border-radius: 10px 0 0 10px;
}

.but {
    width: 20%;
    border: 1px solid rgb(209, 143, 245);
    background: rgb(209, 143, 245);
    border-radius: 0 10px 10px 0;
}

.but:hover {
    background: rgb(195, 99, 247);
}