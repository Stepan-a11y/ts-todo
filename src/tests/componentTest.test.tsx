import { render } from "@testing-library/react";
import AddTask from "../components/AddTask";
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux';

let state: any;
const mockStore = configureStore();
let store;

beforeEach(()=> {
    state = {
        todos: [{
            _id: 0, title: "Some text", starred: false, done: false, editMode: null, date: "16-02-2334"
        },
        {
            _id: 1, title: "Some text 1", starred: true, done: true, editMode: null, date: "18-02-2334"
        },
        {
            _id: 2, title: "Some text 2", starred: false, done: false, editMode: null, date: "22-02-2334"
        }],
        currentPopUpData: { _id: 0, title: "jdgrdjgdjg", starred: null, done: null, editMode: null, date: "", modalActive: false }
    };
});

test('Снапшот кнопки в компоненте AddTask', () => {
        store = mockStore(state);
        const { getByRole } = render(
            <Provider store={store}>
                <AddTask />
            </Provider>
        );

        expect(getByRole('button')).toMatchSnapshot();
});


