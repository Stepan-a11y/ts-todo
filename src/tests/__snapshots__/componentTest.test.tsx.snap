// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Снапшот кнопки в компоненте AddTask 1`] = `
<input
  class="but"
  disabled=""
  type="button"
  value="Добавить"
/>
`;
